src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -311,20 +317,45 @@ else if (SiteUtil.JAVADOC_TOKENS.equals(propertyName)) {
         }
         else {
             final String type = SiteUtil.getType(field, propertyName, currentModuleName, instance);
-            final String relativePathToPropertyTypes =
-                    SiteUtil.getLinkToDocument(currentModuleName, "property_types.xml");
-            final String escapedType = type
-                    .replace("[", ".5B")
-                    .replace("]", ".5D");
-            final String url =
-                    String.format(Locale.ROOT, "%s#%s", relativePathToPropertyTypes, escapedType);
-            sink.link(url);
-            sink.text(type);
-            sink.link_();
+            if ("subset of tokens TokenTypes".equals(type)) {
+                processLinkForTokenTypes(sink);
+            }
+            else {
+                final String relativePathToPropertyTypes =
+                        SiteUtil.getLinkToDocument(currentModuleName, PROPERTY_TYPES_XML);
+                final String escapedType = type
+                        .replace("[", ".5B")
+                        .replace("]", ".5D");
```

Comment:
I remember there was an issue where we should be using some URL building util instead of manually doing this.

Reply:
If we don't do it in this PR, it needs to be added to the other PR.